src/chromium/interoperability-with-cpp/error-handling.md

```diff
@@ -3,39 +3,21 @@
 cxx's support for `Result<T,E>` relies on C++ exceptions, so we can't use that
 in Chromium. Alternatives:
 
-* Where success can be represented as a simple Boolean, as done in our [QR code generator][1]:
-  Return a Boolean representing success, and record results using out-parameters:
-  ```rust,ignore
-  #[cxx::bridge(namespace = "qr_code_generator")]
-  mod ffi {
-    extern "Rust" {
-      fn generate_qr_code_using_rust(
-        data: &[u8],
-        min_version: i16,
-        out_pixels: Pin<&mut CxxVector<u8>>,
-        out_qr_size: &mut usize,
-      ) -> bool;
-    }
-  }
-  ```
-* Where success is more complex, provide a Rust
-  object which can be queried for details of success or failure:
-  ```rust,ignore
-  #[cxx::bridge]
-  mod ffi {
-    extern "Rust" {
-      type PngDecoder;
-      fn create_png_decoder() -> Box<PngDecoder>;
-      fn decode(self: &PngDecoder, png: &[u8]) -> bool; // whether successful
-      fn get_err_code(self: &PngDecoder) -> u32; // or some more complex error type
-      fn get_decoded_image(self: &PngDecoder) -> &[u8];
-            // or some more complex success type
-    }
-  }
```

Comment:
Moved to a separate section: `src/chromium/interoperability-with-cpp/error-handling-png.md`.  Tweaked to more close resemble the PNG prototype (e.g. `is_err` more directly maps to `Result::is_err`;  `unwrap_as_mut` is a combination of `Result::unwrap` and `Result::as_mut`).